lib/grape/validations/validators/length_validator.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
+
+module Grape
+  module Validations
+    module Validators
+      class LengthValidator < Base
+        def initialize(attrs, options, required, scope, **opts)
+          @min = options[:min]
+          @max = options[:max]
+
+          super
+
+          raise ArgumentError, 'min must be an integer greater than or equal to zero' if !@min.nil? && (!@min.is_a?(Integer) || @min.negative?)
+          raise ArgumentError, 'max must be an integer greater than or equal to zero' if !@max.nil? && (!@max.is_a?(Integer) || @max.negative?)
+          raise ArgumentError, "min #{@min} cannot be greater than max #{@max}" if !@min.nil? && !@max.nil? && @min > @max
+        end
+
+        def validate_param!(attr_name, params)
+          param = params[attr_name]
+          param = param.compact if param.respond_to?(:compact)
+
+          raise ArgumentError, "parameter #{param} has an unsupported type. Only strings & arrays are supported" unless params.is_a?(String) || param.is_a?(Array)
```

Comment:
Should this be anything that `.respond_to?(:length)`? 

nit: Let's spell out `&` as `and`

Reply:
Nope I decided on only supporting `String` and `Array` for now since it doesn't make sense to me to support other types e.g. `Hash`.

Reply:
removed the restriction and now allows any type that responds to `#length`.